apps/db-service/src/index.ts

```diff
@@ -109,33 +164,47 @@ const server = net.createServer((socket) => {
         }
 
         // Create a directory for the database
-        await mkdir(dbPath, { recursive: true });
+        await mkdir(dbPath, { recursive: true })
 
         try {
           // Extract the .tar.gz file
-          await pipeline(
-            createReadStream(dumpPath),
-            createGunzip(),
-            extract({ cwd: dbPath })
-          );
+          await pipeline(createReadStream(dumpPath), createGunzip(), extract({ cwd: dbPath }))
         } catch (error) {
-          console.error(error);
-          await rm(dbPath, { recursive: true, force: true }); // Clean up the partially created directory
+          console.error(error)
+          await rm(dbPath, { recursive: true, force: true }) // Clean up the partially created directory
           connection.sendError({
             severity: 'FATAL',
             code: 'XX000',
             message: `Error extracting database: ${(error as Error).message}`,
```

Comment:
Yes the separation would be nice. I usually create a base error class to differential between controlled/uncontrolled error, so it can be routed/formatted to the right system once thrown.
Let's make an issue about that as well. 👍

Reply:
Done here: https://github.com/supabase-community/postgres-new/issues/69